java/client/src/test/java/io/vitess/client/RpcClientTest.java

```diff
@@ -144,6 +150,45 @@ public void setUp() {
     return values;
   }
 
+  /**
+   * waitForVtgateclienttest waits up to 60 seconds for the "vtgateclienttest" binary to come up
+   * and be reachable on the RPC port.
+   *
+   * We use the "GetSrvKeyspace" RPC exemplary to test if vtgate responds.
+   *
+   * @throws SQLException
+   * @throws InterruptedException
+   */
+  private void waitForVtgateclienttest() throws SQLException, InterruptedException {
+    DateTime start = DateTime.now();
+    DateTime deadline = start.plusSeconds(60);
+
+    boolean waited = false;
+    while (DateTime.now().isBefore(deadline)) {
+      try {
+        ctx = Context.getDefault().withDeadlineAfter(Duration.millis(5000)).withCallerId(CALLER_ID);
+        conn.getSrvKeyspace(ctx, "small");
+        // RPC succeeded. Stop testing.
+        break;
+      } catch (SQLTransientException e) {
+        Throwable rootCause = Throwables.getRootCause(e);
+        if (!rootCause.getMessage().contains("Connection refused: ")) {
+          // Non-retryable exception. Fail for good.
+          throw e;
+        }
+
+        System.out.format("Waiting until vtgateclienttest is ready and responds (got exception: %s)\n", rootCause);
+        Thread.sleep(TimeUnit.MILLISECONDS.toMillis(100));
```

Comment:
Ahh, I see.  Unless I'm missing something, this isn't actually changing the behavior of establishing the RPC connection.  It's just `Thread.sleep`-ing for 100 millisecond periods up to 60 seconds, instead of sleeping for a single fixed 10 second period.

If I'm following correctly, then LGTM.  This is more efficient than the fixed 10-second waits... but it sure would be great if the `Context` object's `withDeadlineAfter(...)` value were properly handled, so none of the workarounds would be necessary.